src/import.c

```diff
@@ -55,10 +55,12 @@ gboolean CacheImport(const char *shadow_path, const char *storage_path, const ch
 
   CacheStorage *storage = CacheStorageNew(storage_path);
   if (!CacheStoragePutEntry(storage, username, entry, error)) {
-    // TODO: Log error message
-    g_set_error(error, CACHE_IMPORT_ERROR, CACHE_IMPORT_ERROR_STORAGE_FAIL, "Could not store cache entry.");
     return FALSE;
```

Comment:
You still need to clean memory if you return valse, so you'll probably want to have a variable like "success" with the TRUE/FALSE result, free the memory, then return the variable.